### DISCUSSION TITLE
Why encoding module is made internal?

### DISCUSSION THREAD

**Discussion by sivukhin** [cue-lang/cue#2558] [Q&A] [answered]
I wanted to export CUE definitions to JSON in custom golang tool but unfortunately this is impossible with current structure of CUE source code.

Why `encoding` module is made private?

**Comment by slewiskelly** [comment, marked as answer]
There are other, exported, encoding packages:

- https://pkg.go.dev/cuelang.org/go@v0.6.0/encoding
- https://pkg.go.dev/cuelang.org/go@v0.6.0/pkg/encoding

**Reply by sivukhin** [reply]
Oh, missed that package. Thanks a lot!
